lib/http3/qpack.c

```diff
@@ -556,6 +556,7 @@ static struct st_h2o_qpack_header_t *resolve_dynamic(struct st_h2o_qpack_header_
                                                      const uint8_t **src, const uint8_t *src_end, unsigned prefix_bits,
                                                      const char **err_desc)
 {
+    assert(base_index >= 0);
```

Comment:
Maybe we do not need this, assuming that we add the assertion to `parse_decode_context` as you did in this PR?

Reply:
Thank you, I usually prefer to add asserts to explain the preconditions to the readers, but have such strong aspirations about it. Removed in https://github.com/h2o/h2o/pull/2845/commits/ed2566e671e11baa324178ecdc26cf0831d6ea80